From 2adaf0ceb6c5bfa5cf8677568bff78f09e50c38b Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Wed, 24 Oct 2018 11:07:14 -0500
Subject: [PATCH] Interface.name: decode() string to utf8 before return

Signed-off-by: Haris Okanovic <haris.okanovic@ni.com>

Upstream-Status: Submitted [https://github.com/bat-serjo/PyIface/pull/20]
---
 pyiface/iface.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyiface/iface.py b/pyiface/iface.py
index e8a9043..32ca87a 100644
--- a/pyiface/iface.py
+++ b/pyiface/iface.py
@@ -253,7 +253,7 @@ class Interface(object):
         ifr.data.ifr_ifindex = self._index
         self.__doIoctl(ifr, SIOCGIFNAME)
         self._name = ifr.ifr_name
-        return string_at(self._name)
+        return string_at(self._name).decode('utf8', 'backslashreplace')
 
     @name.setter
     def name(self, val):
-- 
2.19.1

